### DISCUSSION TITLE
Vue Official Plugin Disables Hybrid Mode Due to MylesMurphy.prettify-ts: Seeking Clarification on Auto Detection

### DISCUSSION THREAD

**Discussion by matvejs16** [vuejs/language-tools#4849] [Q&A] [answered]
## Issue Description

I'm working on a project in VSCode and encountered an issue where the "Vue - Official" plugin is automatically disabling Hybrid mode due to the presence of the MylesMurphy.prettify-ts TypeScript plugin. This is causing type errors, particularly with `vue-i18n` and its global `ComponentCustomProperties`.

The Vue Official plugin in VSCode displays the following message:

> Hybrid Mode is disabled automatically because there is a potentially incompatible MylesMurphy.prettify-ts TypeScript plugin installed.

While I can manually enable Hybrid Mode in the settings despite this warning, I prefer to keep the Hybrid Mode setting on "auto". This allows for automatic detection of potential conflicts in future projects, which might be missed if I force it to "true".

## Questions

1. Is this automatic disabling behavior intentional? Is there a known incompatibility between the Vue Official plugin and MylesMurphy.prettify-ts?
2. If it's only a potential incompatibility, is it possible to add MylesMurphy.prettify-ts to a whitelist so that Hybrid Mode remains in "auto" mode without being disabled?
3. Are there any known workarounds to use both plugins together while keeping Hybrid Mode on "auto"?
4. How can users distinguish between potentially harmful conflicts and false positives like this one?

## Environment Details

- VSCode Version: 1.94.0-insider (Commit: f4746c94726e157c9b6534f1355d163ca4efcb38, Date: 2024-09-13T05:04:25.154Z)
- Electron: 30.4.0
- Chromium: 124.0.6367.243
- Node.js: 20.15.1
- V8: 12.4.254.20-electron.0
- Operating System: Linux x64 5.15.0-116-generic

### Plugins
- Vue - Official: v2.1.6
- Prettify TypeScript: Better Type Previews (MylesMurphy.prettify-ts): v0.1.5

### Project Modules
- TypeScript: 5.6.2
- Vue: 3.5.5
- Vue-i18n: 10.0.1

## Additional Information

- Link to MylesMurphy.prettify-ts plugin: https://marketplace.visualstudio.com/items?itemName=MylesMurphy.prettify-ts
- Hybrid Mode can be manually enabled in settings, but this bypasses the auto-detection feature.

## Steps to Reproduce

1. Install Vue Official plugin (v2.1.6) in VSCode
2. Install MylesMurphy.prettify-ts plugin (v0.1.5)
3. Open a Vue project using TypeScript (v5.6.2)
4. Observe that Hybrid Mode is automatically disabled
5. Note that manually enabling Hybrid Mode works, but removes the benefit of auto-detection for future conflicts

I'm seeking a solution that maintains the integrity of the auto-detection feature while allowing the use of the MylesMurphy.prettify-ts plugin. Any insights or guidance on resolving this issue would be greatly appreciated. Thank you for your time and assistance!

**Comment by tsol** [comment]
i support the question!

**Comment by voltuer** [comment]
lol 🤣

**Comment by KazariEX** [comment, marked as answer]
See #5046

**Reply by farzadsoltani** [reply]
Thanks. I can see that it got merged, but I don't think a new version has been released. Do you know when it might get released?
